pkg/planner/core/memtable_predicate_extractor.go

```diff
@@ -332,11 +319,14 @@ func (helper *extractHelper) extractCol(
 		var datums []types.Datum // the memory of datums should not be reused, they will be put into result.
 		switch helper.getStringFunctionName(fn) {
 		case ast.EQ:
-			colName, datums = helper.extractColBinaryOpConsExpr(ctx, extractCols, false, fn)
+			helper.enableScalarPushDown = true
```

Comment:
Since scalarPushDown is only enabled in eq expr, predicates like `lower(table_name) in ("t1", "t2")` cannot be filtered?

Reply:
Yes, this is consistent with previous implementation. If we want to support IN expression, we need to modify `extractColInConsExpr()`.